frontend/src/components/CompaniesList/index.js

```diff
@@ -0,0 +1,143 @@
+import React, {Component} from 'react';
+import {connect} from 'react-redux'
+import {addNewCompany, fetchAllCompanies} from "../../API";
+import Company from "../Company";
+import './index.css'
+import Button from '@material-ui/core/Button';
+import FormControl from '@material-ui/core/FormControl';
+import InputLabel from "@material-ui/core/InputLabel";
+import Input from "@material-ui/core/Input";
+import Box from "@material-ui/core/Box";
+import Table from "@material-ui/core/Table";
+import TableHead from "@material-ui/core/TableHead";
+import TableRow from "@material-ui/core/TableRow";
+import TableCell from "@material-ui/core/TableCell";
+import TableBody from "@material-ui/core/TableBody";
+import {FormGroup} from "@material-ui/core";
+
+
+const mapStateToProps = (state, ownProps) => {
+    return {
+        companies: state.companies
+    }
+};
+
+class CompaniesList extends Component {
+
+    constructor(props) {
+        super(props);
+        this.state = {
+            showAddForm: false,
+            formData: {
+                name: '',
+                city: '',
+                country: '',
+                street: '',
+                postalCode: '',
+                monthlyRent: '',
+            },
+            error: null,
+        }
+    }
+
+    onInputChange(e) {
+        const {id, value} = e.target;
+        this.setState({
+            ...this.state,
+            formData: {
+                ...this.state.formData,
+                [id]: value
+            }
+        })
+    }
+
+    handleSubmit(e) {
+        const {formData} = this.state;
```

Comment:
what happens after submitting form? does not return back to company list view? 
Are you handling API response at component ?

Reply:
Yes, if it resolves with success I just submit form, so we get a new list of companies

Reply:
i meant how does it go back to company list view after submission?

Reply:
There is hook in router component. So it loads data when page reloads

Reply:
ok